gateway/gateway.go

```diff
@@ -12,10 +12,10 @@ import (
 // Status is the state that the client is currently in
 type Status int
 
-// IsConnected returns whether you can send payloads to the Gateway
+// IsConnected returns whether the Gateway is connected
 func (s Status) IsConnected() bool {
 	switch s {
-	case StatusWaitingForGuilds, StatusReady:
+	case StatusWaitingForHello, StatusIdentifying, StatusWaitingForReady, StatusReady:
```

Comment:
It's probably not a good idea to return true for all of these as it may not actually be properly connected. It should only return this for StatusReady. There could be a separate IsSocketConnected() method that tests all of these properties

Reply:
you can simply test for the Status being ready was my idea
as there is only one ready status

Reply:
there is now no usage of it anyway